drop-transfer/src/check.rs

```diff
@@ -7,56 +7,107 @@ use tokio_util::sync::CancellationToken;
 use crate::{protocol, service::State, tasks::AliveGuard, IncomingTransfer, Transfer};
 
 pub(crate) fn spawn(
+    refresh_trigger: tokio::sync::watch::Receiver<()>,
     state: Arc<State>,
     xfer: Arc<IncomingTransfer>,
     logger: Logger,
     guard: AliveGuard,
     stop: CancellationToken,
 ) -> tokio::task::JoinHandle<()> {
     let id = xfer.id();
-    let job = run(state, xfer, logger.clone());
+    let refresh_trigger = refresh_trigger;
 
     tokio::spawn(async move {
         let _guard = guard;
+        loop {
+            let job = run(state.clone(), xfer.clone(), logger.clone());
 
-        tokio::select! {
-            biased;
+            tokio::select! {
+                biased;
 
-            _ = stop.cancelled() => {
-                debug!(logger, "Check job stop for transfer: {id}");
-            },
-            _ = job => (),
+                _ = stop.cancelled() => {
```

Comment:
In general, you should have a different architecture here. There should be a task that runs the `run()` task on channel wakeup in a loop and here the task should be executed inside select with the stop token.
That said the select should have to entries (stop and the task) and shouldn't be in the loop.
The loop should be internal to the task and should not be mixed with the stopping token.

Reply:
adjusted to changes